src/LuaPiGui.cpp

```diff
@@ -338,13 +328,13 @@ static int l_pigui_lineOnClock(lua_State *l)
 	const double hours = LuaPull<double>(l, 2);
 	const double length = LuaPull<double>(l, 3);
 	const double radius = LuaPull<double>(l,4);
-	const ImColor color = LuaPull<ImColor>(l, 5);
+	const Color color = LuaPull<Color>(l,5);
 	const double thickness = LuaPull<double>(l, 6);
 	if (lua_type(l, 1) != LUA_TNIL) {
 		const vector2d center = LuaPull<vector2d>(l, 1);
-		lineOnClock(center, hours, length, radius, color, thickness);
+		lineOnClock(center, hours, length, radius, ImColor(color.r, color.g, color.b, color.a), thickness);
```

Comment:
Please do not do this. You are uselessly duplicating effort that **should** be in the `LuaPull<ImColor>` function instead. I already have to clean up `ImVec2`, please don't add to my workload. Instead, redefine `pi_lua_generic_pull(lua_State*, int, ImColor&)` to pull a `Color` from Lua and convert it to a `ImColor`. The compiler should completely optimize the overhead involved in calling the conversion functions and it's much less work for anyone who later has to implement functions passing Colors to ImGui (i.e. _me_).

Reply:
>  I already have to clean up ImVec2

...Well, probably not a work that must be done...

Anyway, this PR sure will be merged after pull #4576, thus let's talk of this on that PR
and note I simply followed what I suggested there, waiting for answers.

I would also say that these indirections and layers are a possible reason why these table aren't been spotted before, thus my concern.

Reply:
The fact that these were made with tables was most likely due to the fact that it's code that's been around for six years, and nobody had time or willingness to reinvent the wheel at that point.